frame/collective/src/lib.rs

```diff
@@ -355,30 +656,55 @@ impl<T: Trait<I>, I: Instance> Module<T, I> {
 		approved: bool,
 		seats: MemberCount,
 		voting: Votes<T::AccountId, T::BlockNumber>,
-		proposal: T::Hash,
-	) {
+		proposal_hash: T::Hash,
+		proposal: Option<<T as Trait<I>>::Proposal>,
+	) -> Weight {
+		let db = T::DbWeight::get();
+		let mut weight: Weight = 0;
 		if approved {
-			Self::deposit_event(RawEvent::Approved(proposal));
+			Self::deposit_event(RawEvent::Approved(proposal_hash));
 
 			// execute motion, assuming it exists.
-			if let Some(p) = ProposalOf::<T, I>::take(&proposal) {
+			if let Some(p) = proposal {
 				let origin = RawOrigin::Members(voting.threshold, seats).into();
-				let ok = p.dispatch(origin).is_ok();
-				Self::deposit_event(RawEvent::Executed(proposal, ok));
+				let result = p.dispatch(origin);
+				Self::deposit_event(RawEvent::Executed(proposal_hash, result.is_ok()));
+				// TODO: is zero a decent default here?
+				weight = weight.saturating_add(get_result_weight(result).unwrap_or(0)); // P1
```

Comment:
I don't think so, if a dispatchable doesn't refund then this is None AFAIK, so you should use the dispatch info in case of None no ?

Reply:
yeah that's a good default, thank you!